src/Akka.Persistence.Sql/Snapshot/SqlSnapshotStore.cs

```diff
@@ -16,40 +16,73 @@
 
 namespace Akka.Persistence.Sql.Snapshot
 {
-    public class SqlSnapshotStore : SnapshotStore
+    public class SqlSnapshotStore : SnapshotStore, IWithUnboundedStash
     {
         [Obsolete(message: "Use SqlPersistence.Get(ActorSystem).DefaultConfig instead")]
         public static readonly Configuration.Config DefaultConfiguration =
             ConfigurationFactory.FromResource<SqlSnapshotStore>("Akka.Persistence.Sql.snapshot.conf");
 
         private readonly ByteArraySnapshotDao _dao;
+        private readonly ILoggingAdapter _log;
+        private readonly SnapshotConfig _settings;
 
         public SqlSnapshotStore(Configuration.Config snapshotConfig)
         {
+            _log = Context.GetLogger();
+            
             var config = snapshotConfig.WithFallback(SqlPersistence.DefaultSnapshotConfiguration);
-
-            var snapshotConfig1 = new SnapshotConfig(config);
+            _settings = new SnapshotConfig(config);
 
             _dao = new ByteArraySnapshotDao(
-                connectionFactory: new AkkaPersistenceDataConnectionFactory(snapshotConfig1),
-                snapshotConfig: snapshotConfig1,
+                connectionFactory: new AkkaPersistenceDataConnectionFactory(_settings),
+                snapshotConfig: _settings,
                 serialization: Context.System.Serialization,
                 materializer: Materializer.CreateSystemMaterializer((ExtendedActorSystem)Context.System),
                 logger: Context.GetLogger());
+        }
+
+        public IStash Stash { get; set; }
+
+        protected override void PreStart()
+        {
+            base.PreStart();
+            Initialize().PipeTo(Self);
+            BecomeStacked(WaitingForInitialization);
```

Comment:
We're using `BecomeStacked` here because the receive method is sealed in parent class

Reply:
LGTM